sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -97,6 +97,10 @@ case object REPARTITION_BY_NUM extends ShuffleOrigin
 // reader.
 case object REPARTITION_BY_NONE extends ShuffleOrigin
 
+// Indicates that the shuffle operator was not guaranteed the output partitioning so Spark
+// can try to optimize the partition number in AQE framework.
+case object ADAPTIVE_REPARTITION extends ShuffleOrigin
```

Comment:
not sure if `REPARTITION_BY_NONE` should merge with this.

Reply:
yea, sounds like it is overlapping with `REPARTITION_BY_NONE` somehow?